aws-cpp-sdk-core/source/utils/event/EventStreamEncoder.cpp

```diff
@@ -80,19 +80,32 @@ namespace Aws
 
             Aws::Vector<unsigned char> EventStreamEncoder::EncodeAndSign(const Aws::Utils::Event::Message& msg)
             {
-                aws_event_stream_message encoded = Encode(msg);
-                aws_event_stream_message signedMessage = Sign(&encoded);
+                Aws::Vector<unsigned char> outputBits;
 
-                const auto signedMessageLength = signedMessage.message_buffer ? aws_event_stream_message_total_length(&signedMessage) : 0;
+                aws_event_stream_message encoded;
```

Comment:
don't we need to initialize/cleanup the struct?

Reply:
If the `Encode()` function succeeds, then it initializes the struct. I'll rename the function to be more clear

Reply:
I've asked because we have an example in history when CRT was updated to actually access data in the dst buffer:
https://github.com/aws/aws-sdk-cpp/commit/0227ef625c2c53689c61cb9d09eeeea2a78611fb